WordPress/src/main/java/org/wordpress/android/ui/main/MySiteFragment.kt

```diff
@@ -1051,6 +1060,73 @@ class MySiteFragment : Fragment(),
         }
     }
 
+    @Subscribe(sticky = true, threadMode = ThreadMode.MAIN)
+    fun onEventMainThread(event: StorySaveResult) {
+        EventBus.getDefault().removeStickyEvent(event)
+        if (event.isSuccess()) {
+            // TODO WPSTORIES add TRACKS
+            // AnalyticsTracker.track(Stat.MY_SITE_ICON_UPLOAD_UNSUCCESSFUL);
+            // TODO WPSTORIES probably we want to remove this snackbar given we want to immediately start uploading it
+            val snackbarMessage = String.format(
+                    getString(string.story_saving_snackbar_finished_successfully),
+                    getStoryAtIndex(event.storyIndex).title
+            )
+            uploadUtilsWrapper.showSnackbar(
+                    requireActivity().findViewById<View>(R.id.coordinator),
+                    snackbarMessage
+            )
+        } else {
+            // TODO WPSTORIES add TRACKS
+            // AnalyticsTracker.track(Stat.MY_SITE_ICON_UPLOAD_UNSUCCESSFUL);
+            val errorText = String.format(
+                    getString(string.story_saving_snackbar_finished_with_error),
+                    getStoryAtIndex(event.storyIndex).title
+            )
+            val snackbarMessage = buildSnackbarErrorMessage(
+                    requireActivity(),
+                    allErrorsInResult(event.frameSaveResult).size,
+                    errorText
+            )
+            uploadUtilsWrapper.showSnackbarError(
+                    requireActivity().findViewById<View>(R.id.coordinator),
+                    snackbarMessage,
+                    string.story_saving_failed_quick_action_manage,
+                    View.OnClickListener { view: View? ->
+                        val intent = Intent(
+                                requireActivity(),
+                                StoryComposerActivity::class.java
+                        )
+                        intent.putExtra(KEY_STORY_SAVE_RESULT, event)
+                        // TODO add SITE param later when integrating with WPAndroid
+                        // notificationIntent.putExtra(WordPress.SITE, site)
+
+                        // we need to have a way to cancel the related error notification when the user comes
+                        // from tapping on MANAGE on the snackbar (otherwise they'll be able to discard the
+                        // errored story but the error notification will remain existing in the system dashboard)
+                        intent.action = getNotificationIdForError(event.storyIndex).toString() + ""
+
+                        // TODO add NotificationType.MEDIA_SAVE_ERROR param later when integrating with WPAndroid
+                        //        val notificationType = NotificationType.MEDIA_SAVE_ERROR
+                        //        notificationIntent.putExtra(ARG_NOTIFICATION_TYPE, notificationType)
```

Comment:
Do we want to do this integration now? or it's intentional left for a later time. Let me know. 🙏

Reply:
Good question! Yes we need to do that as part of the integration, but it's tackled on a later PR https://github.com/wordpress-mobile/WordPress-Android/pull/12074 👍  (see this [comment](https://github.com/wordpress-mobile/WordPress-Android/pull/12074/files#r434053979) in particular)

Reply:
Oh nice! Thanks!